clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -45,6 +51,26 @@ public HubConnection(String url, Transport transport, Logger logger, boolean ski
             this.logger = new NullLogger();
         }
         this.skipNegotiate = skipNegotiate;
+        this.httpClient = new OkHttpClient.Builder()
+                .cookieJar(new CookieJar() {
```

Comment:
@Tratcher Could you take a look at this cookie storage and see if there are any "gotchas" we should be aware of.

Reply:
Yeah, don't write this code. Find a real implementation.

Reply:
We had a nice chat about this. We'll be using the `Cookie` class from OkHttp and using `matches` and `expiresAt` to return the cookies.